### DISCUSSION TITLE
ESP32-S3-BOX-3 has inverted TFT_RST

### DISCUSSION THREAD

**Discussion by esp32beans** [Bodmer/TFT_eSPI#3002] [New processor/display support or capability request]
The Espressif [ESP32-BOX-S3-3](https://www.espressif.com/en/news/ESP32-S3-BOX-3) has an inverted RST pin. How best to handle this?

Currently, the reset pin 48 is handled in the sketch.

User_Setup.h looks like this (active hardware lines only).

```
#define USER_SETUP_INFO "ESP32-S3-BOX-3"
// Only define one driver, the other ones must be commented out
#define M5STACK // S3-Box-3 uses the same ILI9342 display, don't remove
#define ILI9341_DRIVER       // Generic driver for common displays
#define TFT_BL   47             // LED back-light control pin
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)

// For ESP32-S3-BOX-3
// The hardware SPI can be mapped to any pins

#define TFT_MISO  0
#define TFT_MOSI  6
#define TFT_SCLK  7
#define TFT_CS    5  // Chip select control pin
#define TFT_DC    4  // Data Command control pin
#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST
```

Animated_Eyes_1, TFT_ArcFill, and Cellular_Automata work so far.

**Comment by ophilli** [comment]
I am trying to figure this out myself. I'm starting to think that the reset pin is only an input & shouldn't be driven by the esp itself.

If you look at the [schematic](https://github.com/espressif/esp-box/blob/master/hardware/SCH_ESP32-S3-BOX-3_V1.0/SCH_ESP32-S3-BOX-3-MB_V1.1_20230808.pdf) published by espressif you can see on sheet 6 that the LCD's reset pin is hooked up to Vddd through a resistor.

**Reply by esp32beans** [reply]
LCD_RST is shown on page 3 of the schem. GPIO48 controls LCD_RST. I think the pull up is to ensure the LCD reset is safe during power up before GPIO48 is configured as an output.

See how LCD reset is handled in Espressif's display library.

https://github.com/esp-arduino-libs/ESP32_Display_Panel/blob/2622eeceaa3d6ab7e05bacfe5698be82009aea1c/src/board/esp32_s3_box_3.h#L72C1-L72C1

**Comment by liusida** [comment]
After spending hours, I found this is critical:

```h
#define USE_HSPI_PORT
```

and, yes, we need to manually control TFT_RST, but also, Box-3's TFT_BL is different from Box:
```h
#define TFT_RST  -1  // Reset handled manually in code (ESP-Box-3 inverted logic)
#define TFT_BL  47 // LED back-light
```

And manually reset in .cpp:
```c
{
    pinMode(48, OUTPUT);  // Manual reset pin (ESP-Box-3 BSP)
    digitalWrite(48, HIGH);
    delay(100);
    digitalWrite(48, LOW);
    delay(100);
}
 ```
make sure LOW is for the LCD to work.
